### PR TITLE
Codegen more of our Python arrow serialization

### PR DESCRIPTION
**Pull request by emilk**
### What
This implements codegen of python arrow serialization in cases where we can just delegate to our datatypes.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6154?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6154?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6154)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_types_builder/src/codegen/python/mod.rs [hunk 1]
```diff
@@ -860,77 +860,14 @@ fn code_for_enum(
...
-
-return pa.UnionArray.from_buffers(
-    type=data_type,
-    length=len(data),
-    buffers=buffers,
-    children=children,
-)
-        "##
-    ));
-
```
Comment (emilk) [inline crates/re_types_builder/src/codegen/python/mod.rs:919]: The above code was just moved down to the new `fn quote_arrow_serializetion`

File: rerun_py/rerun_sdk/rerun/datatypes/class_description_map_elem_ext.py [hunk 2]
```diff
@@ -1,50 +0,0 @@
...
-) -> ClassDescriptionMapElem:
-    from . import ClassDescription, ClassDescriptionMapElem
-
-    if isinstance(data, ClassDescriptionMapElem):
-        return data
-    elif isinstance(data, ClassDescription):
-        return ClassDescriptionMapElem(class_id=data.info.id, class_description=data)
-    else:
-        desc = ClassDescription(info=data)
-        return ClassDescriptionMapElem(class_id=desc.info.id, class_description=desc)
```
Comment (emilk) [inline rerun_py/rerun_sdk/rerun/datatypes/class_description_map_elem_ext.py:26]: This was deleted, but all tests still pass so 🤷

File: rerun_py/rerun_sdk/rerun/datatypes/keypoint_pair_ext.py [hunk 3]
```diff
@@ -1,47 +0,0 @@
...
-
-def _keypoint_pair_converter(
-    data: KeypointPairLike,
-) -> KeypointPair:
-    from . import KeypointPair
-
-    if isinstance(data, KeypointPair):
-        return data
-    else:
-        return KeypointPair(*data)
```
Comment (emilk) [inline rerun_py/rerun_sdk/rerun/datatypes/keypoint_pair_ext.py:23]: This was deleted, but all tests still pass so 🤷

**Review by jleibs — APPROVED** [review]
This looks reasonable.

**Comment by Wumpf** [comment]
Jeremy says this is ok and I'm in need for some serialization, so chances this helps me. Merging :)

**Pull request merged**